File: WINGS_AI_Bootcamp/scala_Ymeg/fizz_buzz.scala
// 



@main def fizzBuzz(): Unit =
  def fizzBuzzList(limit: Int): List[String] =
    (1 to limit).map { i =>
      if i % 15 == 0 then "Fizz Buzz"
      else if i % 3 == 0 then "Fizz"
      else if i % 5 == 0 then "Buzz"
      else i.toString
    }.toList

  val result = fizzBuzzList(16)
  println(result)

File: WINGS_AI_Bootcamp/Elixir/fizzBuzz.exs
defmodule FizzBuzz do
  def fb_1(k) when rem(k, 15) == 0 do
    "FIZZBUZZ"
  end

  def fb_1(k) when rem(k, 3) == 0 do
    "FIZZ"
  end

  def fb_1(k) when rem(k, 5) == 0 do
    "BUZZ"
  end

  def fb_1(k) do 
    Integer.to_string(k)
  end
end

res = FizzBuzz.fb_1(15) 
IO.puts(res) 